lib/livebook_web/live/session_live.ex

```diff
@@ -77,6 +78,16 @@ defmodule LivebookWeb.SessionLive do
     end)
   end
 
+  defp assign_policy(socket) do
+    policy =
+      case socket.assigns.live_action do
+        :shared_page -> %{read: true, execute: false, comment: false, edit: false}
+        _ -> %{read: true, execute: true, comment: true, edit: true}
```

Comment:
I would prefer to tackle this slightly differently:

1. We will have a `LivebookWeb.PolicyHook` that will set the policy as an assign

2. Under today's `live_session`, we will declare it as `{LivebookWeb.PolicyHook, :private}` which sets everything to true

3. Our new shared page will have its own `live_session`. In there, we will call `{LivebookWeb.PolicyHook, :shared}`

This way, we keep the policy handling close to the router, which will make it easier for us to see how the rules apply. :)

Reply:
Good!
I did it https://github.com/livebook-dev/livebook/pull/1128/commits/65fc722cda33f32e3c091ffbb262332fffc1300a